### PR TITLE
RELAI Network

### PR DESCRIPTION
**Pull request by morgueye4**
#### RELAI Network

RELAI (Reliable AI) Network is a decentralized platform that aims to help setting foundations to build a secure, transparent, and reproducible environment for AI development and deployment. 

#### Grant [level](https://github.com/w3f/Grants-Program#level_slider-levels)

- [x] **Level 1**:  Up to $10,000, 2 approvals
- [ ] **Level 2**:  Up to $30,000, 3 approvals
- [ ] **Level 3**:  Unlimited, 5 approvals (for >$100k: Web3 Foundation Council approval)

#### Application Checklist

- [x] The [application template](https://github.com/w3f/Grants-Program/blob/master/applications/application-template.md) has been copied and aptly renamed (`project_name.md`).
- [x] I have read the [application guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/Support%20Docs/grant_guidelines_per_category.md).
- [x] Payment details have been provided (bank details via email _or_ BTC, Ethereum (USDC/DAI) or Polkadot/Kusama (USDT) address in the application).
- [x] The software delivered for this grant will be released under an open-source license specified in the application.
- [x] The initial PR contains only one commit (squash and force-push if needed).
- [x] The grant will only be announced once the first milestone [has been accepted](https://github.com/w3f/Grant-Milestone-Delivery#process) (see the [announcement guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/Support%20Docs/announcement-guidelines.md)).
- [ ] I prefer the discussion of this application to take place in a private Element/Matrix channel.

### CODE REVIEW

**Review by semuelle — CHANGES_REQUESTED** [review]
Thanks for the application, @morgueye4. A couple of questions:

- Since the pallet will be the core contribution of this grant, could you add more details about the function(alitie)s and data structures it will implement and how they interact with other components?
- You mention four main actors. Are they differentiated by the system, e.g. through access control?
- In the project details, you mention multiple pallets. If you are planning to fund those through the grant, could you list them in the milestone table as well and add the details I mentioned above?

**Comment by morgueye4** [comment]
@semuelle Thank you for the feedback:

> *Since the pallet will be the core contribution of this grant, could you add more details about the function(alitie)s and data structures it will implement and how they interact with other components?


The MLOrg pallet will be responsible for creating and managing Organizations and access for the users

As for now the following functionalities will be created:
- Create organization
- Add/Remove users 
- Assign roles to users (eg: Admin, AI engineer, MLOps, Stakeholders,…)

The ModelBase pallet will be responsible for managing models and will allow to:

- Create/Update project
- Create/Update model artifact
- Remove model artifact
- Create/Update model version
- Store/Update model experiment
- Create/Update model review

There will be data structures for the : 
Projects, ProjectPurpose, Model, ModelVersion, ModelArtifact, ModelExperiment, ModelReview as well as  Organizations and Roles

Extrinsics cited above will be possible to call accordingly to the role of the user calling. The MLOrg pallet will be loosely coupled to the ModelBase pallet for checking users and roles.


> *You mention four main actors. Are they differentiated by the system, e.g. through access control?

 Indeed the actors will be differentiated by roles and the MLOrg pallet will one part for this. 
  

> * In the project details, you mention multiple pallets. If you are planning to fund those through the grant, could you list them in the milestone table as well and add the details I mentioned above?  

Sure, the application have been updated with the pallets and functionalities list

File: applications/RELAI_Network.md
Comment (keeganquigley) [inline applications/RELAI_Network.md:135]: Thanks for the application. Can you specify which front-end framework you will use here? Also, can you expand on what functionalities the SDK will provide?

**Comment by morgueye4** [comment]
Thank you @keeganquigley 

Angular will be the fronted framework. For the sdk, initially it will support  the following functionalities

- Interacting with the RELAI Network (manage the lifecycle of AI models),
- Identity control and governance
- APIs for getting proof of authenticity for AI model provenance and history.

**Comment by morgueye4** [comment]
HI @randombishop Thanks for asking for clarifications.

> From my understanding, you are requesting a grant to create the first layer for a transparent AI framework, and this layer will only manage organization and model metadata in-chain.

At this stage the layer will manage organizations and model metadata onchain and provide interfaces to check validity of Models (could be thought like the lock sign for https website). Model consumers can easily verify informations about them.

> But where does the data come from and how are the models actually trained?  

At this current stage, the models are trained by the organizations themselves  and use the layer alongside interfacing the web app and/or the SDK to pass inputs(metadata, hashes, proofs,...). 

> Can you please elaborate on how this system will look like after this first layer?

The layer will basically be «A Truth Machine for AI» . It will act like a tether for AI Systems. It will be added  to AI Pipeline stack and runs alongside existing systems for organizations that use it and provide control points for AI. The control points would be.
- Pre-establishing identity and workflow : Blockchain can be used with AI to make sure that the right people, systems, or intelligent agents (participants)—with the right authorization— are the only ones that take part in governance of or modification to the AI. 

- Distributing tamper-evident verification: Blockchain can be used with AI to verify that data and algorithms have not undergone tampering or corruption, 

-  Governing, instructing, and inhibiting intelligent agents : will become very important when wanting to trace or reverse AI, or prove in court that the output of AI is traceable to certain people or organizations. 

- Useful when using branded AI that has underlying components that come from distributed marketplaces. A consumer of AI should be able to test AI to determine whether they trust it before deploying it.

   The system will also serve as foundation for the next phases.     The RELAI Network aims to be a AI hub,  decentralized machine learning, AI compute platform, data storage, AI DAOs, Model Marketplace… So individuals, organizations (centralized or DAOs) that will train their models on it will directly benefits of the system and will have full control on AI lifecycle.

**Review by keeganquigley — COMMENTED** [review]
Thanks for the clarifications @morgueye4 can you add the Angular framework into the application deliverables? As well as the details about the SDK?

**Comment by morgueye4** [comment]
> Thanks for the clarifications @morgueye4 can you add the Angular framework into the application deliverables? As well as the details about the SDK?

Welcome @keeganquigley . The informations have been added and the  on Milestone 2 updated as well

**Comment by morgueye4** [comment]
Welcome @Noc2.

> do you already have some mock-ups or designs which you can share?


I have in this link some raw mockup where it can be found the informations about what will be displayed on the UI with use case diagram and small design about the look and feel of the frontend app. Here is the Link: https://docs.google.com/document/d/17xa2NWYbqphEXV1-Cw1t_IFryk-E3Nui8QuvTTenunA/edit?usp=sharing


> Regarding "validity of Models", how do you ensure this with your solution, and how do you ensure that only the "right people" contribute to a model?

Via the UI each action is allowed accordingly to the control level. To access the web app, the user must login via it's wallet address and informations and accesses are given according to the role of the user (managed by the MLOrg pallet).

**Review by Noc2 — CHANGES_REQUESTED** [review]
Thanks for the update. Could you integrate the [link](https://docs.google.com/document/d/17xa2NWYbqphEXV1-Cw1t_IFryk-E3Nui8QuvTTenunA/edit) into the application?

**Comment by morgueye4** [comment]
> Thanks for the update. Could you integrate the [link](https://docs.google.com/document/d/17xa2NWYbqphEXV1-Cw1t_IFryk-E3Nui8QuvTTenunA/edit) into the application?

Welcome @Noc2 , the link has been added

**Review by Noc2 — CHANGES_REQUESTED** [review]
Thanks for the update. I will mark the application as ready for review and share it with the rest of the team. However, I personally have some concerns here and won't approve it for now: For example, if you only check the access and validity via the UI, I think you can easily attack the system by changing the code of the UI. If the UI instead is somehow encrypted and not open source, you would completely rely on the front-end developer(s). But the rest of the team might have a different opinion here.

**Review by semuelle — COMMENTED** [review]
From a technical point of view, these deliverables do not provide anything new. As for the use case, I am missing proof that this addresses a need. I understand the desire for transparency and reproducibility, but I don't think adding model data to a blockchain addresses this.

**Review by takahser — COMMENTED** [review]
@morgueye4 echoing the sentiments expressed by fellow members of the W3F grants committee, my main concern here is the practicality of the solution you're proposing. Do you have any concrete use cases where a transparent audit trail for ml models would add value?

**Comment by morgueye4** [comment]
@takahser @Noc2 @semuelle I think it will be easier to demonstrate you the utility of this part of the project once the compute / storage layer is built so you can see components works together even though this audit trail layer is independent and can used separately.

I can close the PR for now and may be it can be integrated in other phases to come. Thank you all for your time and feedback.

**Comment by takahser** [comment]
@morgueye4 great to hear, best of luck! Looking forward to seeing a new grant proposal once you've demonstrated the utility of this project with the compute/storage layer in place.

**Pull request closed without merging**
